### ISSUE TITLE
Deploying app hangs at "Launching..."

### ISSUE THREAD

**Issue by simpsora** [deis/deis#3632]
I've got a new app I'm trying to deploy, and it reliably gets stuck at the "Launching..." step.  I've destroyed and recreated the app with no success.  I can deploy other apps into the same cluster without trouble (same deis user, same ssh key).

Here's a gist showing the push, the builder status, and deis-debug-logs:
https://gist.github.com/simpsora/eda7e59d99ae481a170b

It's a small app (20MB).  It pauses for 21 minutes before erroring out.  One interesting thing is that, once it fails, the only mention of it in `fleetctl list-units` is the `cmd` process:

```
discovery-scheduler_v3.cmd.1.service		17d59ace.../10.1.4.82	inactive	dead
```

My `Procfile` doesn't define a `cmd` process, only a `clock` and a `worker`.

Any suggestions for how to fix this?  We have several other apps running in the cluster, and they don't exhibit this problem.

Thanks!

**Comment by bacongobbler** [comment]
Do you have a Dockerfile for your application? The `cmd` process implies that it's not using the Heroku style for building the image, opting for the Docker approach (building from a Dockerfile).

**Comment by simpsora** [comment]
Hi @bacongobbler, no Dockerfile here, just a plain ruby app with a `Procfile`.  I was confused by the `cmd` process as well.  Happy to provide more info if it'll help!

**Comment by bacongobbler** [comment]
If you can reproduce the same issue with https://github.com/deis/example-ruby-sinatra or by posting some sample code that reproduces this then that'd be great :)

**Comment by bacongobbler** [comment]
https://github.com/deis/deis/blob/master/controller/api/models.py#L348-L350 is the relevant issue here

**Comment by carmstrong** [comment]
It's possible that the lack of a `web` process type in your Procfile is confusing Deis, and it's incorrectly detecting a Dockerfile workflow.

**Comment by bacongobbler** [comment]
That's exactly the issue. Thanks for clarifying my mumbling, @carmstrong :)

@simpsora if your app doesn't use a front-end HTTP service, feel free to scale it down to 0 and your clock/worker process types up.

That being said, I think there's some value in updating the logic to catch this specific edge case.

**Comment by simpsora** [comment]
Thanks @bacongobbler, that did it!  When I originally deployed, Deis automatically created and scaled a `cmd` process for me.  To debug, I tried to scale everything down to 0 before deploying, in which case the error still occurs.  If I have the `clock` and `worker` processes scaled up to 1 each, then a deploy works great.

It would be nice if Deis catered for this use case, which is to run scheduled tasks as containers.  I could use a fleet service for this, but I like being able to treat my scheduled processes just like any other deis apps, and use the same toolset to manage/deploy them.

Thanks for your help!

**Comment by bacongobbler** [comment]
> Deis automatically created and scaled a cmd process for me. To debug, I tried to scale everything down to 0 before deploying, in which case the error still occurs

Awesome! You just confirmed https://github.com/deis/deis/issues/3494 which I will be working on to fix this next release :)

**Comment by bacongobbler** [comment]
closed via #3734
